programs/svm-spoke/src/utils/deposit_utils.rs

```diff
@@ -9,3 +9,19 @@ pub fn get_unsafe_deposit_id(msg_sender: Pubkey, depositor: Pubkey, deposit_nonc
 
     keccak::hash(&data).to_bytes()
 }
+
+pub fn derive_delegate_seed_hash(
+    input_token: Pubkey,
+    output_token: Pubkey,
+    input_amount: u64,
+    output_amount: u64,
+    destination_chain_id: u64,
+) -> [u8; 32] {
+    let mut data = Vec::with_capacity(32 + 32 + 8 + 8 + 8);
+    data.extend_from_slice(input_token.as_ref());
+    data.extend_from_slice(output_token.as_ref());
+    data.extend_from_slice(&input_amount.to_le_bytes());
+    data.extend_from_slice(&output_amount.to_le_bytes());
+    data.extend_from_slice(&destination_chain_id.to_le_bytes());
```

Comment:
maybe use `AnchorSerialize::serialize` or `try_to_vec`, so that we don't need to be explicit on individual field encoding